src/main/java/org/jabref/gui/entryeditor/CitationRelationsTab.java

```diff
@@ -0,0 +1,562 @@
+                        filterDifference(fetchedList, observableList, searchType, entry);
+                    }
+                    if (!observableList.isEmpty()) {
+                        listView.refresh();
+                    } else {
+                        Label placeholder = new Label(Localization.lang("No articles found"));
+                        listView.setPlaceholder(placeholder);
+                    }
+                    BooleanBinding booleanBind = Bindings.isEmpty(listView.getCheckModel().getCheckedItems());
+                    importButton.disableProperty().bind(booleanBind);
+                    importButton.setOnMouseClicked(event -> importEntries(listView.getCheckModel().getCheckedItems(), searchType, entry));
+                    setVisibility(true, refreshButton, importButton);
+                })
+                .onFailure(exception -> {
+                    LOGGER.error("Error while fetching citing Articles", exception);
+                    setVisibility(false, abort, progress, importButton);
+                    refreshButton.setVisible(true);
+                    dialogService.notify(exception.getMessage());
+                })
+                .executeWith(Globals.TASK_EXECUTOR);
+        } else {
+            dialogService.notify(Localization.lang("DOI required, Please add DOI to entry before searching."));
+        }
+    }
+
+    /**
+     * Sets visibility of given Nodes to the given value
+     *
+     * @param visibility visibility (true/false)
+     * @param nodes      Nodes to apply new visibility
+     */
+    private void setVisibility(boolean visibility, Node... nodes) {
+        if (nodes != null) {
+            for (Node node : nodes) {
+                node.setVisible(visibility);
+            }
+        }
+    }
+
+    /**
+     * Performs a local lookup of Fields in StandardField.CITING/CITEDBY
+     *
+     * @param entry Current Entry Context
+     * @param field The StandardField to work with
+     */
+    List<BibEntry> runOfflineTask(BibEntry entry, StandardField field) {
+        List<String> keys = getFilteredKeys(entry, field);
+        List<BibEntry> list = new ArrayList<>();
+        LOGGER.info("Current Keys/DOI in " + field.getName() + ":" + keys.toString());
+        for (String key : keys) {
+            Optional<BibEntry> toAdd = getEntryByDOI(key);
+            toAdd.ifPresent(list::add);
+        }
+        return list;
+    }
+
+    /**
+     * Filters an Observable List for Entries, that are already in the operator Field of the Entry.
+     * If the entry is no duplicate, it also add the current entry to the negative operator of the entry in the List
+     *
+     * @param newEntries The List to Filter
+     * @param operator   StandardField.CITING/CITED
+     * @param entry      Current Entry Context
+     */
+    void filterDifference(List<BibEntry> newEntries, ObservableList<CitationRelationItem> observableList, OpenCitationFetcher.SearchType operator, BibEntry entry) {
```

Comment:
Most of the remaining methods should be moved to a new class which is responsible for parsing / serialization of citation info in the bib entries.

Reply:
Where should this Class bei located?

Reply:
Spontaneous answer: Next to `OpenCitationFetcher`? (We should have similar classes there)

Reply:
`logic/citationrelations` would  be another idea